### ISSUE TITLE
Dialyzer error for decode and decode!

### ISSUE THREAD

**Issue by AntoineAugusti** [beatrichartz/csv#125]
## Relevant lines

https://github.com/beatrichartz/csv/blob/29bc8b70be2990e35f44acfe5a8b52cefad9f3c3/lib/csv.ex#L156
https://github.com/beatrichartz/csv/blob/29bc8b70be2990e35f44acfe5a8b52cefad9f3c3/lib/csv.ex#L291

## Problem

At the moment the following code

```elixir
CSV.decode([body], separator: separator, redact_errors: false)
```

will throw a Dialyzer error for `no_return` because the spec doesn't match. The current code expects `decode_options()` **OR** `{:redact_errors, boolean()}`.

I'm not sure how to "extend" typespec with additional arguments or if you want to do another refactor.

**Comment by beatrichartz** [comment]
This has been addressed in `3.2.1` - please reopen if you still encounter issues.

**Comment by AntoineAugusti** [comment]
Thanks! Fixed in https://github.com/beatrichartz/csv/commit/85baadbca611cb01c6eddef809b51eb8e17d9e1d
